cmd/zap/cmd/vector.go

```diff
@@ -0,0 +1,171 @@
+// You may obtain a copy of the License at
+//
+// 		http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build vectors
+// +build vectors
+
+package cmd
+
+import (
+	"encoding/binary"
+	"fmt"
+	"strconv"
+
+	"github.com/RoaringBitmap/roaring"
+	"github.com/blevesearch/go-faiss"
+	"github.com/spf13/cobra"
+)
+
+// storedCmd represents the stored command
+// no big changes here as well.
+var vectorCmd = &cobra.Command{
+	Use:   "vector [path] [field] list <vecID> {optional}",
+	Short: "prints vector index details for a specified field",
+	Long: `The vector command let's you parse a vector section of the specified field and various details about the same.
+	1. check whether a vector section exists for the field.
+	2. if so, fetch the number of vectors, size of the serialized vector index,
+		vector index's config - type, dims, similarity metric.
+	3. given a vector ID, get all the local docNums its present in.
+	4. reconstruct vector given the vectorID.`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		data := segment.Data()
+		pos := segment.FieldsIndexOffset()
+
+		if pos == 0 {
+			// this is the case only for older file formats
+			return fmt.Errorf("file format not supported")
+		}
+
+		// read the number of fields
+		numFields, sz := binary.Uvarint(data[pos : pos+binary.MaxVarintLen64])
+		pos += uint64(sz)
+
+		var fieldID uint64
+		var fieldInv []string
+
+		for fieldID < numFields {
+			addr := binary.BigEndian.Uint64(data[pos : pos+8])
+			fieldSectionMap := make(map[uint16]uint64)
+
+			fieldInv, err := loadFieldData(data, addr, fieldID, fieldSectionMap, fieldInv)
+			if err != nil {
+				return fmt.Errorf("error while parsing the field data %v", err)
+			}
+			if fieldInv[len(fieldInv)-1] == args[1] {
+				vectorSectionOffset, ok := fieldSectionMap[uint16(sectionFaissVectorIndex)]
+				if !ok {
+					return fmt.Errorf("the specified field doesn't have a vector section in it.")
+				}
+				numVecs, indexSize, vecDocIDMap, index, err := decodeSection(data, vectorSectionOffset)
+				if err != nil {
+					return fmt.Errorf("error while decoding the vector section for field %v, err: %v", args[1], err)
+				}
+
+				switch len(args) {
+				case 2:
+					metrics := map[int]string{
+						faiss.MetricInnerProduct:  "inner product",
+						faiss.MetricL2:            "l2 distance",
+						faiss.MetricL1:            "manhattan distance",
+						faiss.MetricLinf:          "chebyshev distance",
+						faiss.MetricLp:            "lp distance",
+						faiss.MetricCanberra:      "canberra distance",
+						faiss.MetricBrayCurtis:    "bray curtis distance",
+						faiss.MetricJensenShannon: "jensen shannon distance",
```

Comment:
We don't support any of these other than first 2, so no need to accommodate these.

Reply:
sorry about that, i think i forgot to save the files before the push :p